indexer/SymbolFormatter.cc

```diff
@@ -117,7 +231,12 @@ SymbolFormatter::getContextSymbol(const clang::DeclContext *declContext) {
       || llvm::isa<clang::ExternCContextDecl>(declContext)) {
     auto decl = llvm::dyn_cast<clang::Decl>(declContext);
     return this->getSymbolCached(decl, [&]() -> std::optional<std::string> {
-      return "c . todo-pkg todo-version ";
+      this->scratchBuffer.clear();
```

Comment:
Since you're returning this string anyway, why not simply allocate one here?

Reply:
1. Use the scratch buffer + copy out of it (current option)
2. Create a new buffer + try to reserve some space based on the implementation details of `SymbolBuilder::formatTo`.
3. Create a new buffer + don't try to reserve any space.

In terms of fiddliness, I'd say option 3 < option 1 < option 2. Option 2 also introduces more coupling.

Since the scratch buffer is getting reused a bunch, it's probably already big enough after it gets used a handful of times. So option 1 will avoid resize+copy operations incurred with option 3. At the same time, option 1 will probably be as good as option 2 in terms of overhead.

One of the main reasons to use the scratch buffer is to avoid redundant resize+copy operations, since lots of symbol strings need to be created per TU. So I think it makes sense to use it for the same purpose here. The main difference is that this use is not behind a neater API like `formatTemporary` (since there's only one use site).